src/main/java/de/tum/cit/aet/artemis/lti/service/LtiDeepLinkingService.java

```diff
@@ -98,23 +129,108 @@ private String buildLtiDeepLinkResponse(String clientRegistrationId, Lti13DeepLi
      * @param courseId    The course ID.
      * @param exerciseIds The set of exercise IDs.
      */
-    private ArrayList<Map<String, Object>> populateContentItems(String courseId, Set<Long> exerciseIds) {
+    private ArrayList<Map<String, Object>> populateExerciseContentItems(String courseId, Set<Long> exerciseIds) {
         ArrayList<Map<String, Object>> contentItems = new ArrayList<>();
         for (Long exerciseId : exerciseIds) {
-            Map<String, Object> item = setContentItem(courseId, String.valueOf(exerciseId));
+            Map<String, Object> item = setExerciseContentItem(courseId, String.valueOf(exerciseId));
             contentItems.add(item);
         }
+        return contentItems;
+    }
 
+    /**
+     * Populate content items for deep linking response with lectures.
+     *
+     * @param courseId   The course ID.
+     * @param lectureIds The set of lecture IDs.
+     */
+    private ArrayList<Map<String, Object>> populateLectureContentItems(String courseId, Set<Long> lectureIds) {
+        ArrayList<Map<String, Object>> contentItems = new ArrayList<>();
+        for (Long lectureId : lectureIds) {
+            Map<String, Object> item = setLectureContentItem(courseId, String.valueOf(lectureId));
+            contentItems.add(item);
+
+        }
         return contentItems;
     }
 
-    private Map<String, Object> setContentItem(String courseId, String exerciseId) {
+    /**
+     * Populate content items for deep linking response with competencies.
+     *
+     * @param courseId The course ID.
+     */
+    private ArrayList<Map<String, Object>> populateCompetencyContentItems(String courseId) {
```

Comment:
this is an ugly return type. Please use List<RecordType> if possible

Reply:
and in general you should avoid using `ArrayList` as Type (except next to the new operator), always use `List` instead